packages/flutter_tools/lib/src/test/flutter_platform.dart

```diff
@@ -576,7 +578,8 @@ import '$testUrl' as test;
 
 void main() {
   print('$_kStartTimeoutTimerMessage');
-  String server = Uri.decodeComponent('$encodedWebsocketUrl');
+  String serverPort = Platform.environment['SERVER_PORT'];
+  String server = Uri.decodeComponent('$encodedWebsocketUrl:$serverPort');
```

Comment:
Just escape the dollar - `\$serverPort`.  There's no need to have it as a param to `_generateTestMain()`

Reply:
done